<?xml version="1.0" encoding="UTF-8"?>
<launch>
  <!--This is only if you have Kinect One; otherwise, ignore and replace Inputs Camera with your device
  <include file="$(find kinect2_bridge)/launch/kinect2_bridge.launch">
    <arg name="publish_tf" value="true"/>
    <arg name="max_depth" value="3.5"/>
  </include>
  -->

  <!--Inputs Camera-->
  <arg name="depth_image" default="/kinect2/hd/image_depth_rect"/>
  <arg name="camera_depth_info" default="/kinect2/hd/camera_info"/>
  <arg name="input_topic" default="/kinect2/hd/image_color"/>
  <arg name="target_frame" default="kinect2_rgb_optical_frame"/>

  <arg name="yolo_model" default="yolov8m-seg.pt"/>
  <arg name="conf_thres" default="0.5"/>
  
  <!--Kinect One Static-->
  <node name="static_kinect" pkg="yolov8" type="static_kinect.py" output="screen"/>

  <!-- Object YoloV8 node-->
  <node name="yolov8_node" pkg="yolov8" type="yolov8_node.py" output="screen">
    <param name="input_topic" value="$(arg input_topic)"/>
    <param name="yolo_model" value="$(arg yolo_model)"/>
    <param name="conf_thres" value="$(arg conf_thres)"/>
  </node>

  <!-- Object detect3D node-->
  <node name="detect3d_node" pkg="yolov8" type="detect3d_node.py" output="screen">
    <param name="depth_image" value="$(arg depth_image)"/>
    <param name="camera_depth_info" value="$(arg camera_depth_info)"/>
    <param name="conf_thres" value="$(arg conf_thres)"/>
    <param name="target_frame" value="$(arg target_frame)"/>
  </node>

  <!-- Object debug node -->
  <node name="debug_node" pkg="yolov8" type="debug_node.py" output="screen">
    <param name="input_topic" value="$(arg input_topic)"/>
  </node>

  <node pkg="rviz" type="rviz" name="rviz" args="-d $(find yolov8)/rviz/labrob.rviz"/>
</launch>
